gcloud/datastore/test_dataset.py

```diff
@@ -157,6 +157,11 @@ def allocate_ids(self, dataset_id, key_pbs):
         num_pbs = len(key_pbs)
         return [_KeyProto(i) for i in range(num_pbs)]
 
+    def delete_entities(self, dataset_id, key_pbs):
```

Comment:
Why is this method here?  Wouldn't it be easier / clearer to define a `_Connection` class in `test_key` which had only the methods we need there?